cEP-0010.md

```diff
@@ -46,11 +46,15 @@ Here is the list of aspects to be implemented along with their corresponding
   |--- Formatting
   |    |--- Quotation
   |    |    |--- preferred_quotation
+  |    |    |--- escape_delimiter
+  |    |    |--- alternate_delimiter_language
   |    |--- Length
   |    |    |--- LineLength
   |    |    |    |--- max_line_length
   |    |    |--- FileLength
```

Comment:
`FileLength` feels very odd in a `Formatting` aspect.

It is a part of defining what is a 'line', to determine what should be counted for *metrics*.  The reason to restrict the lines of code per file is to prevent complexity growing infinitely.

'FileLength' should IMO be part of `Smell` aspect.

SLOC per file is a quick estimation of module complexity.

Compare with `LineLength` which can be fixed by re-formatting the line, changing symbols/tokens (e.g. adding line feeds, brackets) but not changing any semantics, ..

`FileLength` often can not be solved without a re-write, especially where the file contains only one public class, and the language does not allow splitting a single class across multiple files (Python, Java, etc)

Very interested in other peoples opinions on this.

If we agree it is 'wrong' to call this a formatting aspect, we should raise an issue about moving it, and then separately decide whether it is in scope for this GSoC project or not.

Reply:
I think `FileLength` has been part of `Formatting` aspect because the related bear `LineCountBear` simply counts the number of lines in a file (with no complexity measurement). If the linecount is more than expected, a little formatting can resolve it very well. e.g. using a semicolon in c to convert two statements into one. See https://github.com/coala/coala-bears/blob/d83ddce283a58326074b57810dc3aec84e30e8e8/bears/general/LineCountBear.py#L14.

However, if we are planning to implement line counting algorithms as tastes to `FileLength` aspect and stretching the same functionality for the bear too, then I totally agree that it should be part of `Smell` aspect (perhaps `FileLength` as a subaspect of `Smell.Complexity` suits better). Here it is https://github.com/coala/coala/blob/454b565d0b38890694c87479aa3b6ecc1c7b9296/coalib/bearlib/aspects/Smell.py#L511.

I agree with you that it should be moved via a separate issue. And so I am removing `min_file_length` and `include_blank_lines` from this cEP.

And IMO `FileLength` is out of the scope of this project as it is a total redesign of the bear (not forgetting the aspect part) including the research of line-counting algorithms. It can be a separate project by itself. I have already 5 bears listed (excluding this one) which very well satisfy my project requirements.